client/blocks/jitm/index.jsx

```diff
@@ -1,52 +1,125 @@
 /**
  * External Dependencies
  */
-import React from 'react';
-import { connect } from 'react-redux';
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect, useDispatch } from 'react-redux';
+import { assign } from 'lodash';
+import debugFactory from 'debug';
 
 /**
  * Internal dependencies
  */
+import config from 'config';
+import TrackComponentView from 'lib/analytics/track-component-view';
+import { recordTracksEvent } from 'state/analytics/actions';
 import { getSelectedSite } from 'state/ui/selectors';
 import { getTopJITM } from 'state/jitm/selectors';
-import { dismissJetpackJITM } from 'state/jitm/actions';
-import Banner from 'components/banner';
-
-export const JITM = ( {
-	currentSite,
-	hasJitm,
-	callToAction,
-	description,
-	featureClass,
-	id,
-	message,
-	onDismiss,
-} ) =>
-	hasJitm &&
-	currentSite && (
-		<Banner
-			callToAction={ callToAction }
-			title={ message }
-			description={ description }
-			disableHref
-			dismissPreferenceName={ featureClass + '123' }
-			dismissTemporary={ true }
-			onDismiss={ onDismiss( currentSite.ID, id, featureClass ) }
-			event={ `jitm_nudge_click_${ id }` }
-			href={ `https://jetpack.com/redirect/?source=jitm-${ id }&site=${ currentSite.domain }` }
-			target={ '_blank' }
-		/>
+import { dismissJITM, setupDevTool } from 'state/jitm/actions';
+import AsyncLoad from 'components/async-load';
+import QueryJITM from 'components/data/query-jitm';
+import 'state/data-layer/wpcom/marketing';
+
+const debug = debugFactory( 'calypso:jitm' );
+
+function renderTemplate( template, props ) {
+	if ( template === 'notice' ) {
+		return <AsyncLoad { ...props } require="blocks/jitm/templates/notice" />;
+	}
+
+	if ( template === 'sidebar-banner' ) {
+		return <AsyncLoad { ...props } require="blocks/jitm/templates/sidebar-banner" />;
+	}
+
+	return <AsyncLoad { ...props } require="blocks/jitm/templates/default" />;
+}
+
+function getEventHandlers( props, dispatch ) {
+	const { jitm, currentSite, messagePath } = props;
+	const tracks = jitm.tracks || {};
+	const eventProps = { id: jitm.id, jitm: true };
+	const handlers = {};
+
+	if ( tracks.display ) {
+		handlers.trackImpression = () => (
+			<TrackComponentView
+				eventName={ tracks.display.name || 'calypso_jitm_nudge_impression' }
+				eventProperties={ assign( {}, tracks.display.props, eventProps ) }
```

Comment:
Nitpick: This `assign()` call and the followings can be replaced by `{ ...tracks.display.props, ... eventProps}` , and then we can remove `assign()` import completely.